File: c/gpt/p14.c
// p14: function prompts user for index, returns -1 if out of bounds; program returns 0
#include <stdio.h>

int getValuePrompt(void)
{
    int id_sequence[] = {10, 20, 30, 40};
    int n = sizeof(id_sequence) / sizeof(id_sequence[0]);
    int idx;
    printf("Enter index: ");
    if (scanf("%d", &idx) != 1)
        return -1;
    if (idx < 0 || idx >= n)
        return -1;
    return id_sequence[idx];
}

int main(void)
{
    int val = getValuePrompt();
    printf("%d\n", val);
    return 0;
}

File: java/gpt/P14.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class P14 {
    public static int getValuePrompt() throws Exception {
        int[] id_sequence = {10, 20, 30, 40};
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Enter index: ");
        String s = br.readLine();
        int idx;
        try { idx = Integer.parseInt(s.trim()); } catch (Exception e) { return -1; }
        if (idx < 0 || idx >= id_sequence.length) return -1;
        return id_sequence[idx];
    }

    public static void main(String[] args) throws Exception {
        int val = getValuePrompt();
        System.out.println(val);
        System.exit(0);
    }
}
